src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -411,41 +358,45 @@ private SafeProcessHandle GetProcessHandle()
         /// <summary>Get the minimum and maximum working set limits.</summary>
         private void GetWorkingSetLimits(out IntPtr minWorkingSet, out IntPtr maxWorkingSet)
         {
-            SafeProcessHandle handle = null;
-            try
+            using (SafeProcessHandle handle = GetProcessHandle(Interop.Advapi32.ProcessOptions.PROCESS_QUERY_INFORMATION))
             {
-                handle = GetProcessHandle(Interop.Advapi32.ProcessOptions.PROCESS_QUERY_INFORMATION);
-
-                // GetProcessWorkingSetSize is not exposed in coresys.  We use 
-                // GetProcessWorkingSetSizeEx instead which also returns FLAGS which give some flexibility
-                // in terms of Min and Max values which we neglect.
                 int ignoredFlags;
                 if (!Interop.Kernel32.GetProcessWorkingSetSizeEx(handle, out minWorkingSet, out maxWorkingSet, out ignoredFlags))
-                {
                     throw new Win32Exception();
-                }
-            }
-            finally
-            {
-                ReleaseProcessHandle(handle);
             }
         }
 
         private string GetMainWindowTitle()
         {
             IntPtr handle = MainWindowHandle;
-            if (handle == (IntPtr)0)
-            {
+            if (handle == IntPtr.Zero)
                 return string.Empty;
-            }
 
-            int length = Interop.User32.GetWindowTextLength(handle) * 2;
+            int length = Interop.User32.GetWindowTextLengthW(handle);
 
             if (length == 0)
+            {
+#if DEBUG
+                // We never used to throw here, want to surface possible mistakes on our part
+                int error = Marshal.GetLastWin32Error();
+                Debug.Assert(error == 0, $"Failed GetWindowTextLengthW(): { new Win32Exception(error).Message }");
+#endif
                 return string.Empty;
+            }
 
             StringBuilder builder = new StringBuilder(length);
-            Interop.User32.GetWindowText(handle, builder, builder.Capacity);
+            length = Interop.User32.GetWindowTextW(handle, builder, builder.Capacity + 1);
+
+            if (length == 0)
+            {
+#if DEBUG
+                // We never used to throw here, want to surface possible mistakes on our part
+                int error = Marshal.GetLastWin32Error();
+                Debug.Assert(error == 0, $"Failed GetWindowTextW(): { new Win32Exception(error).Message }");
+#endif
+            }
+
+            builder.Length = length;
```

Comment:
> Doesn't the StringBuilder marshaler account for that, though?

I does in that it creates a native buffer of Capacity +1, but that doesn't help in passing the "correct" capacity to the API (which are almost always assuming the null goes into the count you passed).

The off-by-one is all the more painful given that an out buffer is 3x capacity in allocations plus the ToString() allocation